### PR TITLE
Ignore bad optionID = "0" when translating invoice_lines

### LINKED ISSUE #3611: Handle trans_line with optionID: "0"
Don't fail integrate such records

### PR DESCRIPTION
**Pull request by clemens-msupply**
Fixes #3611 

# 👩🏻‍💻 What does this PR do? 
Ignores bad optionId data.

### CODE REVIEW

File: server/service/src/sync/test/test_data/invoice_line.rs [hunk 1]
```diff
@@ -386,7 +386,7 @@ const TRANS_LINE_OM_UNSET_TAX_FIELDS: (&str, &str) = (
         "manufacturer_ID": "",
         "medicine_administrator_ID": "",
         "note": "every FOUR to SIX hours when necessary ",
-        "optionID": "",
+        "optionID": "0",
```
Comment (clemens-msupply) [inline server/service/src/sync/test/test_data/invoice_line.rs:389]: Added this to the existing test case

**Comment by roxy-dao** [comment]
Postgres tests are failing =[

**Comment by clemens-msupply** [comment]
> Postgres tests are failing =[

Thanks fixed it, this was actually because of the report_* views...

The postgres tests are a bit flaky would need to look into it at some point on why they are failing sometimes...

**Comment by roxy-dao** [comment]
> > Postgres tests are failing =[
> 
> Thanks fixed it, this was actually because of the report_* views...
> 
> The postgres tests are a bit flaky would need to look into it at some point on why they are failing sometimes...

Still seems to be failing for me

**Review by roxy-dao — APPROVED** [review]
LGTM~

**Pull request merged**